e2e/test/AppContextTests.cs

```diff
@@ -45,28 +49,60 @@ public async Task AppConfig_ReceiveAsyncAfterDispose_Sasl_Amqp()
 
             if (isFlagSet)
             {
-                await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+                await ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only, false).ConfigureAwait(false);
+            }
+            else
+            {
+                Assert.Fail("AppContext flag DisableObjectDisposedExceptionForReceiveAsync not set");
+            }
+        }
+
+        [TestMethod]
+        public async Task AppConfig_ReceiveAsyncWithCancellationTokenAfterDispose_Sasl_Amqp()
+        {
+
+            bool isFlagSet = false;
+            bool flag = false;
+
+#if !NET451
+            isFlagSet = AppContext.TryGetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, out flag);
```

Comment:
This should though work in NET451: can we manually test that having this added in app.config takes effect when running on .NET Framework 4.7 (or greater)?